lib/matplotlib/backends/backend_qt5agg.py

```diff
@@ -75,7 +75,13 @@ def paintEvent(self, e):
         In Qt, all drawing should be done inside of here when a widget is
         shown onscreen.
         """
-        # FigureCanvasQT.paintEvent(self, e)
+        # if the canvas does not have a renderer, then
+        # FigureCanvasAgg.draw(self)  has never been called so force it to
+        # be called to create the renderer and to render the initial view
+        # of the figure
+        if not hasattr(self, 'renderer'):
+            FigureCanvasAgg.draw(self)
```

Comment:
This is probably unsafe. paintEvent is usually called by a different thread, not from main thread the matplotlib code is running in. Calling draw here may cause concurrent access to the matplotlib figure, which is not thread safe.